### PR TITLE
Standardise prometheus metrics

### PR DESCRIPTION
**Pull request by leonerd**
When I originally wrote the process-wide metric collection, I wasn't aware that there are standard names and semantics for the set of metrics you're supposed to export. Namely:
  https://prometheus.io/docs/instrumenting/writing_clientlibs/#standard-and-runtime-collectors

This PR attempts to address this by:
 * Providing currently-duplicate exporting of process-wide stats that already existed, now to standard names and units (`process_cpu_*_seconds_total`, `process_open_fds`)
 * Adding new standard process-wide metrics that didn't used to exist (`process_*_memory_bytes`, `process_start_time_seconds`, `process_max_fds`)

The intention is that once there's about a week of data in the new format logged by prometheus, I'll update grafana graphs to use those standard metric names instead, then make another commit to remove the "old" variable names.

### CODE REVIEW

**Comment by erikjohnston** [comment]
Why are we using `/proc/self/stat` rather than using `rusage`?

**Comment by leonerd** [comment]
Because `/proc/self/stat` has a bunch of other metrics we want that `rusage` doesn't give us anyway

**Comment by erikjohnston** [comment]
I think at this point all the globals should be wrapped up into an actual class or split out into a separate module (or both)

File: synapse/metrics/__init__.py [hunk 1]
```diff
@@ -139,12 +144,27 @@ def update_resource_metrics():
...
+def update_resource_metrics():
+    global rusage
+    rusage = getrusage(RUSAGE_SELF)
+
+    if HAVE_PROC_SELF_STAT:
+        global stats
+        with open("/proc/self/stat") as s:
+            line = s.read()
+            # line is PID (command) more stats go here ...
+            stats = line.split(") ", 1)[1].split(" ")
```
Comment (erikjohnston) [inline synapse/metrics/__init__.py]: Can we parse the stats here, rather than requiring the users of stat to know the magic numbers?

**Comment by leonerd** [comment]
> I think at this point all the globals should be wrapped up into an actual class or split out into a separate module (or both)

Yes, perhaps I'll pull the lot out into a separate `ProcessMetrics` class - is here in this file OK or put it in a new one?

**Comment by erikjohnston** [comment]
Depends on the size really, if it gets a bit big then moving to a separate file would probably be sensible, but if it small then leaving it there is probably fine.

**Comment by leonerd** [comment]
Actually turned out that moving it into a new file was easier/neater than into a separate class within the same file.

File: synapse/metrics/__init__.py (hunk 1 above)
Comment (leonerd) [inline synapse/metrics/__init__.py]: Done 6453d03

**Comment by leonerd** [comment]
Three Jenkins failures appear unrelated. A single flakey test looks like:

    408 - Can add global push rule for override

**Comment by erikjohnston** [comment]
LGTM

**Pull request merged**